docs/user-guide/themes.rst

```diff
@@ -0,0 +1,92 @@
+.. _userguide_themes:
+
+Plotting Themes
+===============
+
+PyVista plotting parameters can be controlled on a plot by plot basis
+or through a global theme, making it possible to control mesh colors
+and styles through one global configuration.
+
+.. jupyter-execute::
+   :hide-code:
+
+   # using ipyvtk as it loads faster
+   import pyvista
+   pyvista.set_jupyter_backend('static')
+
+The default theme parameters in PyVista can be accessed and displayed with:
+
+.. jupyter-execute::
+
+    import pyvista
+    pyvista.global_theme
+
+Default plotting parameters can be accessed individually by their
+attribute names:
+
+.. jupyter-execute::
+
+   pyvista.global_theme.color
+
+Here's an example plot of the Stanford Dragon using default plotting
+parameters:
+
+.. jupyter-execute::
+
+    from pyvista import examples
+    dragon = examples.download_dragon()
+    dragon.plot(cpos='xy')
+
+These parameters can then be modified globally with:
+
+.. jupyter-execute::
+
+   pyvista.global_theme.color = 'red'
+   pyvista.global_theme.background = 'white'
+   pyvista.global_theme.axes.show = False
+
+Now, the mesh will be plotted with the new global parameters:
+
+.. jupyter-execute::
+
+    dragon.plot(cpos='xy')
+
+This is identical to plotting the mesh with the following parameters:
+
+.. code:: python
+
+   dragon.plot(cpos='xy', color='red', background='white', show_axes=False)
+
+
+Creating A Custom Theme
+-----------------------
+You can customize a theme based on one of the built-in themes and then
+apply it globally with:
+
+.. jupyter-execute::
+
+   # create a theme based off the DocumentTheme
+   my_theme = pyvista.themes.DocumentTheme()
+   my_theme.cmap = 'jet'
+   my_theme.show_edges = True
```

Comment:
This had a typo (`.show_edge`) which makes me think we might want to define a custom `__setattr__` that raises when new attributes are defined. Might get a bit messy because we'd have to define the allowed attributes on a per-subclass basis. Perhaps we can store a private list of allowed attribute names for each subclass, and define the `__setattr__` on the base.

Reply:
Good point.  For fun, I wanted to see how ``numpy`` did it:
```py

>>> arr = numpy.ones(10)
>>> arr.abc = 123

AttributeError: 'numpy.ndarray' object has no attribute 'abc'
```
Turns out they're using `__slots__`.  See https://stackoverflow.com/questions/3603502/prevent-creating-new-attributes-outside-init

Reply:
This is one of those things that's cool to know and 100% not something I would have researched on my own.

Reply:
I also thought about `__slots__` but then I realized that I wasn't sure how that interacted with inheritance. If it works it's perfect.

Reply:
Inheritance is interesting with `__slots__`.  Effectively, if what you're inheriting has `__dict__`, `__slots__` is ignored.  So, all base classes must have `__slots__` for this to work, hence why it's empty and added to `_ThemeConfig`.